app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderToVerifyApplyAndRemoveCouponCodeTest.xml

```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminCreateOrderToVerifyApplyAndRemoveCouponCodeTest">
+        <annotations>
+            <stories value="Create Order with offline payment methods"/>
+            <title value="Create Order to verify apply and remove coupon code test"/>
+            <description value="Create Order to verify apply and remove coupon code test"/>
+            <severity value="CRITICAL"/>
+            <testCaseId value="MC-37386"/>
+            <group value="sales"/>
+        </annotations>
+        <before>
+            <!-- Login as admin -->
+            <actionGroup ref="AdminLoginActionGroup" stepKey="LoginAsAdmin"/>
+
+            <!-- Create customer -->
+            <createData entity="Simple_US_Customer" stepKey="createCustomer"/>
+
+            <!-- Create simple product -->
+            <createData entity="SimpleProduct2" stepKey="createSimpleProduct">
+                <field key="price">10</field>
+            </createData>
+
+            <!-- Create sales rule with coupon -->
+            <createData entity="SalesRuleSpecificCouponWithFixedDiscount" stepKey="createCartPriceRule"/>
+            <createData entity="SimpleSalesRuleCoupon" stepKey="createCouponForCartPriceRule">
+                <requiredEntity createDataKey="createCartPriceRule"/>
+            </createData>
+
+            <!-- Enable Bank Transfer payment -->
+            <magentoCLI command="config:set {{EnablePaymentBankTransferConfigData.path}} {{EnablePaymentBankTransferConfigData.value}}" stepKey="enableBankTransferPayment"/>
+
+            <!--Set default flat rate shipping method settings-->
+            <magentoCLI command="config:set {{EnableFlatRateConfigData.path}} {{EnableFlatRateConfigData.value}}" stepKey="enableFlatRate"/>
```

Comment:
Don't we need to revert this state as well on after node?

Reply:
I think that in this case you can use the following 

<createData entity="FlatRateShippingMethodDefault" stepKey="setDefaultFlatRateShippingMethod"/>

as used in 
https://github.com/magento/magento2/blob/1ee91ca2d9c6650c83f4b5f6abc1983fa1fef629/app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithSimpleProductTest.xml

https://github.com/magento/magento2/blob/1ee91ca2d9c6650c83f4b5f6abc1983fa1fef629/app/code/Magento/Sales/Test/Mftf/Test/AdminHoldCreatedOrderTest.xml


@gabrieldagama I don't see the revert of this flatrate set in other tests. Is this a good practice in this case or is not required?

Reply:
@shikhamis11 You can use the files I mentioned as a reference for before actions.

Reply:
@rodrigowebjump Yeah, ideally we should leave the application state as we found before the test ran, the reason for that is basically to avoid flaky tests. Other tests may rely on the "default" application state to pass successfully, and that may bring problems for us on PR processing.